vulnerabilities/forms.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+from django import forms
+
+from vulnerabilities.models import Package, PackageRelatedVulnerability, Vulnerability
+
+
+def get_package_types():
+    pkg_types = [(i.type, i.type) for i in Package.objects.distinct("type").all()]
+    pkg_types.append((None, "package type"))
+    return pkg_types
+
+
+def get_package_namespaces():
+    pkg_namespaces = [
+        (i.namespace, i.namespace)
```

Comment:
I am puzzled by the repetition.

Reply:
https://docs.djangoproject.com/en/3.1/ref/forms/fields/#django.forms.ChoiceField.choices 

Example:  For the tuple ("x","1")  the user will be able to select "1" in the frontend which will be interpreted as  "x" in the backend.   

In this case I want the user to select namespace for what it is in the frontend as well as backend. This whole  thing is done because I want to use `ChoiceField` instead of `CharField` which is provided by the `ModelForm`.